### PR TITLE
Add more REST API docs (Dashboard Widget Data, Dynamic Content, Point Actions, Point Triggers, Webhooks)

### PR DESCRIPTION
**Pull request by dennisameling**
Almost there! Another bunch of REST API docs as mentioned in https://github.com/mautic/developer-documentation-new/issues/97

### CODE REVIEW

**Comment by dennisameling** [comment]
The remaining Vale errors are a known issue with not being able to do `vale off` on headings: https://github.com/errata-ai/vale/issues/340#issuecomment-1339799018

File: docs/rest_api/dynamic_content.rst [hunk 1]
```diff
@@ -0,0 +1,430 @@
+Dynamic Content
+###############
+
+Use this endpoint to obtain details on Mautic's web Dynamic Content.
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:4]:
```suggestion
Use this endpoint to obtain details on Mautic's Dynamic Web Content.
```

File: docs/rest_api/dynamic_content.rst [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+   <?php
+
+   $data = array(
+       'name'        => 'Dynamic Content A',
+       'isPublished' => 1
+   );
+
+   $dynamicContent = $dynamicContentApi->create($data);
+
+Create a new dynamicContent.
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:223]:
```suggestion
Create a new Dynamic Content.
```
I _think_ these should be separate words unless it's purposely written as above in the code sample.

File: docs/rest_api/dynamic_content.rst [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+       'name'        => 'New dynamicContent name',
+       'isPublished' => 0
+   );
+
+   // Create new a dynamicContent of ID 1 isn't found?
+   $createIfNotFound = true;
+
+   $dynamicContent = $dynamicContentApi->edit($id, $data, $createIfNotFound);
+
+Edit a new dynamicContent. Note that this supports PUT or PATCH depending on the desired behavior.
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:317]:
```suggestion
Edit a new Dynamic Content. Note that this supports PUT or PATCH depending on the desired behavior.
```
As above.

File: docs/rest_api/dynamic_content.rst [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+
+**PUT** creates a dynamicContent if the given ID doesn't exist and clears all the Dynamic Content information, adds the information from the request.
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:319]:
```suggestion
**PUT** creates a Dynamic Content if the given ID doesn't exist and clears all the Dynamic Content information, adds the information from the request.
```

File: docs/rest_api/dynamic_content.rst [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+**PATCH** fails if the Dynamic Content with the given ID doesn't exist and updates the Dynamic Content field values with the values from the request.
+
+.. vale off
+
+**HTTP Request**
+
+.. vale on
+
+To edit a dynamicContent and return a 404 if the Dynamic Content isn't found:
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:328]:
```suggestion
To edit a Dynamic Content and return a 404 if the Dynamic Content isn't found:
```

File: docs/rest_api/dynamic_content.rst [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+
+``PATCH /dynamiccontents/ID/edit``
+
+To edit a dynamicContent and create a new one if the Dynamic Content isn't found:
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:332]:
```suggestion
To edit a Dynamic Content and create a new one if the Dynamic Content isn't found:
```

File: docs/rest_api/dynamic_content.rst [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+
+.. vale on
+
+.. code-block:: php
+
+   <?php
+
+   $dynamicContent = $dynamicContentApi->delete($id);
+
+Delete a dynamicContent.
```
Comment (RCheesley) [inline docs/rest_api/dynamic_content.rst:414]:
```suggestion
Delete a Dynamic Content.
```

File: docs/rest_api/point_actions.rst [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+
+.. list-table::
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Description
+   * - ``id``
+     - int
+     - ID of the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:88]:
```suggestion
     - ID of the Point Action
```

I think this should use the full term.

File: docs/rest_api/point_actions.rst [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``name``
+     - string
+     - Name of the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:91]:
```suggestion
     - Name of the Point Action
```

File: docs/rest_api/point_actions.rst [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``description``
+     - string/null
+     - Description of the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:94]:
```suggestion
     - Description of the Point Action
```

File: docs/rest_api/point_actions.rst [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+     - Category name
+   * - ``type``
+     - string
+     - Point Action type
+   * - ``isPublished``
+     - boolean
+     - Published state
+   * - ``publishUp``
+     - datetime/null
+     - Point publish date/time
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:106]:
```suggestion
     - Point Action publish date/time
```

File: docs/rest_api/point_actions.rst [hunk 12]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``publishDown``
+     - datetime/null
+     - Point unpublish date/time
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:109]:
```suggestion
     - Point Action unpublish date/time
```

File: docs/rest_api/point_actions.rst [hunk 13]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``dateAdded``
+     - ``datetime``
+     - Point creation date/time
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:112]:
```suggestion
     - Point Action creation date/time
```

File: docs/rest_api/point_actions.rst [hunk 14]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``createdBy``
+     - int
+     - ID of the User that created the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:115]:
```suggestion
     - ID of the User that created the Point Action
```

File: docs/rest_api/point_actions.rst [hunk 15]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``createdByUser``
+     - string
+     - Name of the User that created the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:118]:
```suggestion
     - Name of the User that created the Point Action
```

File: docs/rest_api/point_actions.rst [hunk 16]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``dateModified``
+     - datetime/null
+     - Date/time point was last modified
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:121]:
```suggestion
     - Date/time Point Action was last modified
```

File: docs/rest_api/point_actions.rst [hunk 17]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``modifiedBy``
+     - int
+     - ID of the User that last modified the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:124]:
```suggestion
     - ID of the User that last modified the Point Action
```

File: docs/rest_api/point_actions.rst [hunk 18]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``modifiedByUser``
+     - string
+     - Name of the User that last modified the point
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:127]:
```suggestion
     - Name of the User that last modified the Point Action
```

File: docs/rest_api/point_actions.rst [hunk 19]
```diff
@@ -0,0 +1,388 @@
...
+   * - ``delta``
+     - int
+     - The number of Points to assign to the Contact when executing this action
```
Comment (RCheesley) [inline docs/rest_api/point_actions.rst:130]:
```suggestion
     - The number of Points to assign to the Contact when executing this Point Action
```

File: docs/rest_api/point_triggers.rst [hunk 20]
```diff
@@ -0,0 +1,505 @@
...
+
+.. list-table::
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Description
+   * - ``id``
+     - int
+     - ID of the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:95]:
```suggestion
     - ID of the Point Trigger
```

File: docs/rest_api/point_triggers.rst [hunk 21]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``name``
+     - string
+     - Name of the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:98]:
```suggestion
     - Name of the Point Trigger
```

File: docs/rest_api/point_triggers.rst [hunk 22]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``description``
+     - string/null
+     - Description of the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:101]:
```suggestion
     - Description of the Point Trigger
```

File: docs/rest_api/point_triggers.rst [hunk 23]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``category``
+     - string
+     - Category name
+   * - ``isPublished``
+     - boolean
+     - Published state
+   * - ``publishUp``
+     - datetime/null
+     - Point publish date/time
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:110]:
```suggestion
     - Point Trigger publish date/time
```

File: docs/rest_api/point_triggers.rst [hunk 24]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``publishDown``
+     - datetime/null
+     - Point unpublish date/time
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:113]:
```suggestion
     - Point Trigger unpublish date/time
```

File: docs/rest_api/point_triggers.rst [hunk 25]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``dateAdded``
+     - ``datetime``
+     - Point creation date/time
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:116]:
```suggestion
     - Point Trigger creation date/time
```

File: docs/rest_api/point_triggers.rst [hunk 26]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``createdBy``
+     - int
+     - ID of the User that created the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:119]:
```suggestion
     - ID of the User that created the Point Trigger
```

File: docs/rest_api/point_triggers.rst [hunk 27]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``createdByUser``
+     - string
+     - Name of the User that created the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:122]:
```suggestion
     - Name of the User that created the Point Trigger
```

File: docs/rest_api/point_triggers.rst [hunk 28]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``dateModified``
+     - datetime/null
+     - Date/time point was last modified
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:125]:
```suggestion
     - Date/time Point Trigger was last modified
```

File: docs/rest_api/point_triggers.rst [hunk 29]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``modifiedBy``
+     - int
+     - ID of the User that last modified the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:128]:
```suggestion
     - ID of the User that last modified the Point Trigger
```

File: docs/rest_api/point_triggers.rst [hunk 30]
```diff
@@ -0,0 +1,505 @@
...
+   * - ``modifiedByUser``
+     - string
+     - Name of the User that last modified the point
```
Comment (RCheesley) [inline docs/rest_api/point_triggers.rst:131]:
```suggestion
     - Name of the User that last modified the Point Trigger
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
A few fairly small suggestions - thanks so much for porting this over @dennisameling 🚀

**Review by RCheesley — APPROVED** [review]

**Pull request merged**